src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs

```diff
@@ -337,6 +360,51 @@ PooledHashSet<ISymbol> GetCandidateSymbolsReferencedInDocComments(INamedTypeSymb
                 return builder;
             }
 
+            ArrayBuilder<string> GetDebuggerDisplayAttributeArguments(INamedTypeSymbol namedTypeSymbol)
+            {
+                var builder = ArrayBuilder<string>.GetInstance();
+                AddDebuggerDisplayAttributeArguments(namedTypeSymbol, builder);
+                return builder;
+            }
+
+            void AddDebuggerDisplayAttributeArguments(INamedTypeSymbol namedTypeSymbol, ArrayBuilder<string> builder)
+            {
+                AddDebuggerDisplayAttributeArgumentsCore(namedTypeSymbol, builder);
+
+                foreach (var member in namedTypeSymbol.GetMembers())
+                {
+                    switch (member)
+                    {
+                        case INamedTypeSymbol nestedType:
+                            AddDebuggerDisplayAttributeArguments(nestedType, builder);
+                            break;
+
+                        case IPropertySymbol property:
+                        case IFieldSymbol field:
+                            AddDebuggerDisplayAttributeArgumentsCore(member, builder);
+                            break;
+                    }
+                }
+            }
+
+            void AddDebuggerDisplayAttributeArgumentsCore(ISymbol symbol, ArrayBuilder<string> builder)
+            {
+                foreach (var attribute in symbol.GetAttributes())
+                {
+                    if (attribute.AttributeClass == _debuggerDisplayAttributeType &&
+                        attribute.ConstructorArguments.Length == 1 &&
+                        attribute.ConstructorArguments[0].Kind == TypedConstantKind.Primitive &&
+                        attribute.ConstructorArguments[0].Type.SpecialType == SpecialType.System_String)
+                    {
+                        var value = attribute.ConstructorArguments[0].Value as string;
```

Comment:
up to you, but a cute thing you can do is: ```&& attribute.ConstructorArguments[0] is var arg && arg.Kind ...```  That way you don't have to repeat the indexing 3 times.